### PR TITLE
Add support for multiple base installations of same Fmw

### PR DESCRIPTION
**Pull request by hajee**
The current version of fmw, doesn't support the installation of the same middleware more than once. If you try, you get duplicate resources.  If however you want to have a middleware home for SOA and one for BAM and apply different patches to these, you'll need to do just that. This PR add's support for installing multiple homes with the same version. You will have to use different titles to the fmw to do this.  After this you can use opatch to differentiate the middleware homes

### CODE REVIEW

File: spec/defines/fmw_spec.rb [hunk 1]
```diff
@@ -69,11 +69,11 @@
                   :source               => '/mnt',
                   :temp_directory       => '/tmp',
                 }}
-    let(:title) {'My Oracle-"SOA" Installation!?'}
```
Comment (alexjfisher) [inline spec/defines/fmw_spec.rb]: This spec test was added in https://github.com/biemond/biemond-orawls/commit/d75151ea03e571c767c0fb3c900314ebc5b392cd

The resource title was interesting on purpose. :)
The context is 'resource title needing sanitising'
https://github.com/biemond/biemond-orawls/blob/master/spec/defines/fmw_spec.rb#L56

File: manifests/fmw.pp [hunk 2]
```diff
@@ -82,15 +82,15 @@
       $oracleHome = $oracle_home_dir
     }
 
-    $createFile1 = "${download_dir}/${fmw_product}/Disk1"
+    $createFile1 = "${download_dir}/${name}/Disk1"
```
Comment (alexjfisher) [inline manifests/fmw.pp]: instead of ${name}, do you want to use ${sanitised_title} instead?

File: spec/defines/fmw_spec.rb (hunk 1 above)
Comment (hajee) [inline spec/defines/fmw_spec.rb]: I'll spice it up again. ;-)

**Comment by hajee** [comment]
@alexjfisher I've fixed all the stuff you mentioned

**Pull request merged**